File: 118-pascals-triangle/118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        
        ans.add(Arrays.asList(1));
        if(numRows ==1){
            return ans;
        }
        
        for(int i=1;i<numRows;i++){
            List<Integer> arr = new ArrayList<>();
            for(int j=0;j<=i;j++){
                if(j==0 || j==i){
                    arr.add(1);
                }
                else{
                    int a = ans.get(i-1).get(j-1);
                    int b = ans.get(i-1).get(j);
                    arr.add(a+b);
                }
            }
            ans.add(arr);
        }
        return ans;
    }
}

File: 118-pascals-triangle/118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>ans;
         ans.push_back({1});
        if(numRows==1){
            return ans;
            ;
        }
        
        for(int i=1; i<numRows; i++){
            vector<int>res;
            for(int j =0; j<=i;j++){
                if(j==0 || j==i){
                res.push_back(1);
                }
                else{
                    // int a = res[i-1].push_back(j-1);
                    // int b = res[i-1].push_back(j);
                    // ans.push_back(res[i-1].push_back(ans)+res[i-1].push_back(j));
                    res.push_back(ans[i-1][j-1]+ ans[i-1][j]);
                }
            }
            ans.push_back(res);
        }
        return ans;
    }
};